packages/central-server/__tests__/sync/PatientMergeSync.test.js

```diff
@@ -0,0 +1,256 @@
+    };
+
+    mergeEnc = await models.Encounter.create({
+      ...fake(Encounter),
+      ...baseEncounter,
+      patientId: merge.id,
+    });
+    mergeEnc2 = await models.Encounter.create({
+      ...fake(Encounter),
+      ...baseEncounter,
+      patientId: merge.id,
+    });
+    await models.Encounter.create({
+      ...fake(Encounter),
+      ...baseEncounter,
+      patientId: keep.id, // keep encounter
+    });
+
+    await PatientFacility.upsert({
+      id: PatientFacility.generateId(),
+      patientId: keep.id,
+      facilityId: facility.id,
+    });
+    await PatientFacility.upsert({
+      id: PatientFacility.generateId(),
+      patientId: merge.id,
+      facilityId: facility.id,
+    });
+  };
+
+  beforeAll(async () => {
+    ctx = await createTestContext();
+    ({ models } = ctx.store);
+  });
+
+  afterAll(() => ctx.close());
+
+  beforeEach(async () => {
+    await models.Facility.truncate({ cascade: true, force: true });
+    await models.PatientFacility.truncate({ cascade: true, force: true });
+    await models.Encounter.truncate({ cascade: true, force: true });
+    await models.Note.truncate({ cascade: true, force: true });
+    await models.Patient.truncate({ cascade: true, force: true });
+    await models.LocalSystemFact.truncate({ cascade: true, force: true });
+    await models.Department.truncate({ cascade: true, force: true });
+    await models.Location.truncate({ cascade: true, force: true });
+    await models.User.truncate({ cascade: true, force: true });
+  });
+
+  it('pulls child records (notes) of merged encounters after merging patients', async () => {
+    const { LocalSystemFact } = models;
+
+    await setupMergeData();
+
+    const note = await models.Note.create({
+      ...fake(models.Note),
+      recordId: mergeEnc.id,
+      recordType: NOTE_RECORD_TYPES.ENCOUNTER,
+    });
+    const note2 = await models.Note.create({
+      ...fake(models.Note),
+      recordId: mergeEnc2.id,
+      recordType: NOTE_RECORD_TYPES.ENCOUNTER,
+    });
+
+    const NEW_SYNC_TICK = '4';
+    await LocalSystemFact.set(FACT_CURRENT_SYNC_TICK, NEW_SYNC_TICK);
+
+    await mergePatient(models, keep.id, merge.id);
+
+    const centralSyncManager = initializeCentralSyncManager();
+    const { sessionId } = await centralSyncManager.startSession();
+
+    await centralSyncManager.updateLookupTable();
+
+    // Start the snapshot for pull process
+    await centralSyncManager.setupSnapshotForPull(
+      sessionId,
+      {
+        since: NEW_SYNC_TICK - 1, // get all changes since the last sync tick
+        facilityIds: [facility.id],
+        deviceId: facility.id,
+      },
+      () => true,
+    );
+
+    const changes = await centralSyncManager.getOutgoingChanges(sessionId, {
+      limit: 10,
+    });
+    const notes = changes.filter((c) => c.recordType === 'notes');
+    expect(notes).toHaveLength(2);
+    expect(notes.map((n) => n.recordId).sort()).toEqual([note.id, note2.id].sort());
```

Comment:
I know this test basically covers all child record types of encounters, but it might be worth adding one more, like `lab_requests` or something. Notes sometimes ends up getting handled differently due to the polymorphic relationships, wouldn't want to do something with notes in future and break one of the other tables.